src/main/java/com/exadel/discountwebapp/discount/mapper/DiscountMapper.java

```diff
@@ -65,9 +80,17 @@ public void updateEntity(DiscountRequestVO request, Discount discount) {
     }
 
     private void provideDiscountDependencies(DiscountRequestVO request, Discount discount) {
-        Vendor vendor = vendorRepository.findById(request.getVendorId()).orElse(null);
-        Category category = categoryRepository.findById(request.getCategoryId()).orElse(null);
+        var vendor = vendorRepository.findById(request.getVendorId())
+                .orElseThrow(() -> new EntityNotFoundException("Vendor not found"));
+        var category = categoryRepository.findById(request.getCategoryId())
+                .orElseThrow(() -> new EntityNotFoundException("Category not found"));
+
+        List<Location> locations = request.getLocationIds().stream()
+                .map(locationId -> locationRepository.findById(locationId)
```

Comment:
Access to the database should not be performed in a loop. Because every iteration generates a separate SQL query. Let's optimize it by fetching all locations by location ids in just one SQL query

Reply:
Replaced to findAllById(). Hope good or there can be better solution. If so, please let me know :)

Reply:
yes, there can be a better solution) but let's move to the optimizations later